packages/gatsby/src/utils/page-data.ts

```diff
@@ -184,6 +188,10 @@ export async function flush(parentSpan?: Span): Promise<void> {
     // them, a page might not exist anymore щ（ﾟДﾟщ）
     // This is why we need this check
     if (page) {
+      if (page.path && nodeManifestPagePathMap[page.path]) {
```

Comment:
Instead of an object we should either use a Map or maybe even better use the Gatsby cache so we don't have to hold it in memory. `cache.set(`manifest-page-${pagePath}`, manifestId)`.